crates/hyperdrive-math/src/short/max.rs

```diff
@@ -433,13 +425,14 @@ impl State {
     ) -> Result<FixedPoint> {
         let checkpoint_exposure_shares =
             FixedPoint::try_from(checkpoint_exposure.max(I256::zero()))? / self.vault_share_price();
-        // FIXME: Check this against solidity; should we be using solvency here or just share reserves?
-        // solvency = share_reserves - long_exposure / vault_share_price - min_share_reserves
-        let solvency = self.calculate_solvency();
-        let guess = self.vault_share_price() * (solvency + checkpoint_exposure_shares);
+        let guess = self.vault_share_price().mul_down(
+            self.share_reserves() + checkpoint_exposure_shares
+                - self.long_exposure().div_down(self.vault_share_price())
+                - self.minimum_share_reserves(),
```

Comment:
Did you update this so that it's more obvious what the components are? `calculate_solvency` should just be `self.share_reserves() - self.long_exposure().div_down(self.vault_share_price()) - self.minimum_share_reserves()` but maybe I'm missing something.

Reply:
I double checked because I had forgotten the motivation here. The calculation was equivalent, and in general I am more of a fan of centralizing functions like `calculate_solvency`. I made this change to make it easier to directly compare to the solidity function `calculateMaxShortGuess`.